### PR TITLE
Fix CI failure dict assertion with deepdiff

### PR DESCRIPTION
**Pull request by gopidesupavan**
https://github.com/apache/airflow/actions/runs/12471950302/job/34810119013#step:7:2722

Hope order is not important? 

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/api_fastapi/core_api/routes/ui/test_structure.py [hunk 1]
```diff
@@ -404,7 +405,7 @@ class TestStructureDataEndpoint:
     def test_should_return_200(self, test_client, params, expected):
         response = test_client.get("/ui/structure/structure_data", params=params)
         assert response.status_code == 200
-        assert response.json() == expected
+        assert not DeepDiff(response.json(), expected, ignore_order=True)
```
Comment (amoghrajesh) [inline tests/api_fastapi/core_api/routes/ui/test_structure.py:408]: Can we just sort by key instead?

Comment (gopidesupavan) [reply tests/api_fastapi/core_api/routes/ui/test_structure.py:408]: Problem is its nested map of array, its not working with sort_keys.

Comment (gopidesupavan) [reply tests/api_fastapi/core_api/routes/ui/test_structure.py:408]: or else we need to build sort function and sort all the nested keys then assert, dont think its worth here? DeepDiff is simple to assert

Comment (amoghrajesh) [reply tests/api_fastapi/core_api/routes/ui/test_structure.py:408]: Yeah looks cool

Comment (amoghrajesh) [reply tests/api_fastapi/core_api/routes/ui/test_structure.py:408]: Why rebuild the wheel!

Comment (gopidesupavan) [reply tests/api_fastapi/core_api/routes/ui/test_structure.py:408]: BTW DeepDiff internally doing sorting to compare.

**Review by amoghrajesh — APPROVED** [review]
Order shouldnt matter imo.

**Comment by potiuk** [comment]
Nice!

File: tests/api_fastapi/core_api/routes/ui/test_structure.py (hunk 1 above)
Comment (potiuk) [reply tests/api_fastapi/core_api/routes/ui/test_structure.py:408]: Nice small tool.

BTW. @gopidesupavan -> maybe we could use `deephash` from the same set of tools to make stable hashes of config dictionaries? 

There is one thing that is "specific" to how config help is extracted (Python 3.12 performs slightly different dedenting of docstrings extracted from methods) - but I think `deephash` should nicely handle the last change with opts sorting and possibly other in-stabilities that `click` might introduce - see https://github.com/apache/airflow/pull/45156

Comment (gopidesupavan) [reply tests/api_fastapi/core_api/routes/ui/test_structure.py:408]: Yeah it helps they have some useful tools. https://zepworks.com/deepdiff/current/. 

Yes deepdiff really useful and nice tool. we use this tool more in our work to compare terraform plans(its a huge nested dict :) ). with old state and new states change to extract some data for triage.

**Pull request merged**